src/sc/sidechain.cpp

```diff
@@ -713,17 +713,30 @@ int CSidechain::getNumBlocksForScFeeCheck()
 
 int CSidechain::getMaxSizeOfScFeesContainers()
 {
-    int epochLength = fixedParams.withdrawalEpochLength;
-    int numBlocks = getNumBlocksForScFeeCheck();
-
-    //assert(epochLength > 0);
-    if (epochLength == 0) {
-        maxSizeOfScFeesContainers = 1;
-    } else {
-        maxSizeOfScFeesContainers = numBlocks / epochLength;
-        if (maxSizeOfScFeesContainers == 0 || (numBlocks % epochLength != 0) )
-        {
-            maxSizeOfScFeesContainers++;
+    if (maxSizeOfScFeesContainers == -1) {
+        maxSizeOfScFeesContainers = getNumBlocksForScFeeCheck();
+
+        if (!isNonCeasing()) {
+            int epochLength = fixedParams.withdrawalEpochLength;
+            int numBlocks = getNumBlocksForScFeeCheck();
+            if (epochLength == 0) {
+                maxSizeOfScFeesContainers = 1;
+            } else {
+                maxSizeOfScFeesContainers = numBlocks / epochLength;
```

Comment:
Is this safe ? Any chance `epochLength` assumes a negative value ?

Reply:
Non negativity of epoch length is checked in `Sidechain::checkTxSemanticValidity()` before sidechain creation, hence in the current configuration `epochLenght` cannot be negative. Things may change if in the future we would allow `epochLenght` to assume negative values for managing special cases. I propose putting a preemptive assert.

Reply:
Assertion added in f0bab99cd7559a9bab4c8dc65103df6eea2b4b76